tmt/steps/execute/upgrade.py

```diff
@@ -28,19 +27,45 @@
 
 
 @dataclasses.dataclass
-class ExecuteUpgradeData(tmt.steps.execute.ExecuteStepData):
-    url: Optional[str] = None
-    upgrade_path: Optional[str] = None
+class ExecuteUpgradeData(ExecuteInternalData):
+    url: Optional[str] = field(
+        default=cast(Optional[str], None),
+        option=('-u', '--url'),
+        metavar='REPOSITORY',
+        help='URL of the git repository with upgrade tasks.')
+    upgrade_path: Optional[str] = field(
+        default=cast(Optional[str], None),
+        option=('-p', '--upgrade-path'),
+        metavar='PLAN_NAME',
+        help='Upgrade path corresponding to a plan name in the repository with upgrade tasks.')
 
     # Inherit from tmt.steps.discover.fmf.DiscoverFmfStepData
-    ref: Optional[str] = None
-    filter: Optional[List[str]] = None
-    test: Optional[List[str]] = None
-    exclude: Optional[List[str]] = None
-
-    _normalize_test = tmt.utils.NormalizeKeysMixin._normalize_string_list
-    _normalize_filter = tmt.utils.NormalizeKeysMixin._normalize_string_list
-    _normalize_exclude = tmt.utils.NormalizeKeysMixin._normalize_string_list
+    ref: Optional[str] = field(
+        default=cast(Optional[str], None),
+        option=('-r', '--ref'),
+        metavar='REVISION',
+        help='Branch, tag or commit specifying the git revision.')
+    filter: List[str] = field(
```

Comment:
Seems that the `test` and `filter` options are swapped here?

Reply:
Hmmm, it's a pitty that tests did not catch this.

Reply:
Fixed in 57fc488a.